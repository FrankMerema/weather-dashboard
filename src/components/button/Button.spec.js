import { mount, shallowMount } from '@vue/test-utils';
import Button from './Button';

describe('Button', () => {
  test('is a Vue component', () => {
    const wrapper = mount(Button);
    expect(wrapper.isVueInstance()).toBeTruthy();
  });
});

describe('Button snapshot', () => {
  test('render a disabled button', () => {
    const wrapper = shallowMount(Button, {
      propsData: {
        disabled: true
      },
      slots: {
        default: 'Disabled button'
      }
    });

    expect(wrapper.element).toMatchSnapshot();
  });

  test('render a submit button', () => {
    const wrapper = shallowMount(Button, {
      propsData: {
        type: 'submit'
      },
      slots: {
        default: 'Submit button'
      }
    });

    expect(wrapper.element).toMatchSnapshot();
  });
});
