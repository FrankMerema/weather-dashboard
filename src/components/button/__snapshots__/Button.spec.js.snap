// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button snapshot render a disabled button 1`] = `
<button
  class="btn"
  disabled="disabled"
  type="button"
>
  Disabled button
</button>
`;

exports[`Button snapshot render a submit button 1`] = `
<button
  class="btn"
  type="submit"
>
  Submit button
</button>
`;
